src/lib/resolvers/emoji.ts

```diff
@@ -0,0 +1,30 @@
+import { EmojiRegex, TwemojiRegex } from '@sapphire/discord-utilities';
+import { Snowflake, Util } from 'discord.js';
+import { Identifiers } from '../errors/Identifiers';
+import { err, ok } from '../parsers/Result';
+
+export function resolveEmoji(parameter: string) {
+	const twemoji = TwemojiRegex.exec(parameter)?.[0] ?? null;
+
+	if (twemoji) {
+		return ok(twemoji);
+	}
+
+	const emojiId = EmojiRegex.exec(parameter)?.groups?.id;
+
+	if (emojiId) {
+		const resolved = Util.parseEmoji(emojiId) as GuildEmojiStructure;
+
+		if (resolved) {
+			return ok(resolved);
+		}
+	}
+
+	return err(Identifiers.ArgumentEmojiError);
+}
+
+export interface GuildEmojiStructure {
+	animated: boolean;
+	name: string;
+	id: Snowflake;
```

Comment:
```suggestion
	animated?: boolean;
	name: string | null;
	id: string | null;
```

Following my other comment, so it can be used for twemojis. Also change `id` to `string`. In DiscordJS / discord-api-types there have been many shifts between `Snowflake` and `string` and eventually they just decided to go for `string`, so lets follow that.